.ci/no_old_refs.sh

```diff
@@ -5,27 +5,66 @@ set -e
 # Script requires GITHUB_TOKEN env variable
 MENTIONED_ISSUES=/tmp/mentioned_issues
 CLOSED_ISSUES=/tmp/failed_issues
+LINKED_ISSUES_FORMATTED=/tmp/linked_issues
+
+# Linked issues that are mentioned in the code
+LINKED_ISSUES_MENTIONED=/tmp/linked_issues_mentioned
 API_GITHUB_PREFIX="https://api.github.com/repos"
 GITHUB_HOST="https://github.com"
+CHECKSTYLE_ISSUE_PREFIX="https:\/\/github.com\/checkstyle\/checkstyle\/issues\/"
+MAIN_REPO="checkstyle/checkstyle"
+DEFAULT_BRANCH="master"
+
+# These are modified when event is of type pull_request
+if [ ! -z "$PR_HEAD_REPO_NAME" ]; then
+  MAIN_REPO=$PR_HEAD_REPO_NAME
+  DEFAULT_BRANCH=$GITHUB_HEAD_REF
+fi
 
 # collect issues where full link is used
-grep -IPohr "(after|[Tt]il[l]?) $GITHUB_HOST/[\w.-]+/[\w.-]+/issues/\d{1,5}" . \
-  | sed -e 's/.*github.com\///' >> $MENTIONED_ISSUES
+grep -IPonr "(after|[Tt]il[l]?) $GITHUB_HOST/[\w.-]+/[\w.-]+/issues/\d{1,5}" . \
+  | perl -pe 's/:(?!\d).*github.com\//:/' >> $MENTIONED_ISSUES
 
 # collect checkstyle issues where only hash sign is used
-grep -IPohr "[Tt]il[l]? #\d{1,5}" . \
-  | sed -e 's/.*#/checkstyle\/checkstyle\/issues\//' >> $MENTIONED_ISSUES
+grep -IPonr "[Tt]il[l]? #\d{1,5}" . \
+  | perl -pe 's/:(?!\d).*#/:checkstyle\/checkstyle\/issues\//' >> $MENTIONED_ISSUES
+
+# $LINKED_ISSUES need formatting before the are used
+if [ ! -z "$LINKED_ISSUES" ]; then
+  echo $LINKED_ISSUES | sed -e 's/,/\n/g' >> $LINKED_ISSUES_FORMATTED
+  sed -i "s/^/$CHECKSTYLE_ISSUE_PREFIX/g" $LINKED_ISSUES_FORMATTED
+fi
+
+for MENTIONED_ISSUES_LINES in $(sort -u $MENTIONED_ISSUES); do
```

Comment:
What for sorting is required ?

Reply:
It isn't required at all, previously it was required as we were just reporting the issue number and we do not wanted duplicates, no it isn't required, removed.

Reply:
It isn't required at all, previously it was required as we were just reporting the issue number and we do not wanted duplicates, no it isn't required, removed.